api/storage/file.go

```diff
@@ -72,3 +81,23 @@ func (o *fileImpl) Delete() error {
 
 	return nil
 }
+
+func (o *fileImpl) PresignUrl(mode Mode) (string, error) {
+	op := v1.StoragePreSignUrlRequest_READ
```

Comment:
have updated to take an options object, which takes expiry as well (that was missing). options contains a validation method that will error on invalid mode provided (arbitrary int). It will still be read by default if a mode is not provided.